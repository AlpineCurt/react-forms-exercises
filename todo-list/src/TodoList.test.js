import React from "react";
import { render, fireEvent } from "@testing-library/react"
import Todolist from "./TodoList";

test("renders without crashing", () => {
    render(<Todolist />);
});

test("matches snapshot", () => {
    const {asFragment} = render(<Todolist />);
    expect(asFragment()).toMatchSnapshot();
});

test("add new todo item", () => {
    const {getByLabelText, queryByText} = render(<Todolist />);
    const todoInput = getByLabelText("New Todo");
    const btn = queryByText("Add Todo");
    expect(queryByText("X")).not.toBeInTheDocument();
    fireEvent.change(todoInput, {target: {value: "Eat Tofu"}});
    fireEvent.click(btn);
    expect(queryByText("Eat Tofu")).toBeInTheDocument();
});

test("remove a todo item", () => {
    const {getByLabelText, queryByText} = render(<Todolist />);
    const todoInput = getByLabelText("New Todo");
    const btn = queryByText("Add Todo");
    expect(queryByText("X")).not.toBeInTheDocument();
    fireEvent.change(todoInput, {target: {value: "Eat Tofu"}});
    fireEvent.click(btn);
    expect(queryByText("Eat Tofu")).toBeInTheDocument();
    const removeBtn = queryByText("X");
    fireEvent.click(removeBtn);
    expect(queryByText("Eat Tofu")).not.toBeInTheDocument();
});