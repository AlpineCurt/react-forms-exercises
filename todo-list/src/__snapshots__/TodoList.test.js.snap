// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h3>
    Add a Todo
  </h3>
  <form>
    <label
      for="form-todo"
    >
      New Todo
    </label>
    <input
      id="form-todo"
      name="todo"
      placeholder="Todo"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
  <h3>
    Todos:
  </h3>
  <ul
    class="TodoList"
  />
</DocumentFragment>
`;
